### PR TITLE
Improve & unify docs for dirichlet priors

### PR DESCRIPTION
**Pull request by jonaschn**
Make documentation for alpha & eta consistent for AuthorTopicModel, LdaModel and LdaMulticore.

### CODE REVIEW

File: gensim/models/atmodel.py [hunk 1]
```diff
@@ -279,23 +291,17 @@ def __init__(self, corpus=None, num_topics=100, id2word=None, author2doc=None, d
...
         self.eta, self.optimize_eta = self.init_dir_prior(eta, 'eta')
-
-        self.random_state = utils.get_random_state(random_state)
-
         assert (self.eta.shape == (self.num_terms,) or self.eta.shape == (self.num_topics, self.num_terms)), (
             "Invalid eta shape. Got shape %s, but expected (%d, 1) or (%d, %d)" %
             (str(self.eta.shape), self.num_terms, self.num_topics, self.num_terms)
         )
 
+        self.random_state = utils.get_random_state(random_state)
```
Comment (mpenkov) [inline gensim/models/atmodel.py:303]: Why is this necessary in a documentation-only PR?

Or does this PR address more than just documentation?

Comment (jonaschn) [reply gensim/models/atmodel.py:303]: This PR is not documentation-only but mainly (only some minor refactoring).

File: gensim/models/atmodel.py [hunk 2]
```diff
@@ -279,23 +291,17 @@ def __init__(self, corpus=None, num_topics=100, id2word=None, author2doc=None, d
         self.init_empty_corpus()
 
         self.alpha, self.optimize_alpha = self.init_dir_prior(alpha, 'alpha')
-
         assert self.alpha.shape == (self.num_topics,), \
             "Invalid alpha shape. Got shape %s, but expected (%d, )" % (str(self.alpha.shape), self.num_topics)
 
-        if isinstance(eta, str):
-            if eta == 'asymmetric':
-                raise ValueError("The 'asymmetric' option cannot be used for eta")
```
Comment (jonaschn) [inline gensim/models/atmodel.py:288]: I moved this check into `init_dir_prior` (from `LdaModel`).

File: gensim/models/atmodel.py [hunk 3]
```diff
@@ -279,23 +291,17 @@ def __init__(self, corpus=None, num_topics=100, id2word=None, author2doc=None, d
...
-
         self.eta, self.optimize_eta = self.init_dir_prior(eta, 'eta')
-
-        self.random_state = utils.get_random_state(random_state)
```
Comment (jonaschn) [inline gensim/models/atmodel.py:292]: I moved this statement behind the `assert` statements.
This is more consistent with the previous checks (see alpha prior) before.

Comment (mpenkov) [reply gensim/models/atmodel.py:292]: Ah, sorry, I misread the diff.

**Comment by mpenkov** [comment]
Thank you @jonaschn !
